core/src/main/resources/error/error-classes.json

```diff
@@ -153,13 +215,24 @@
     "message" : [ "Unsupported literal type %s %s" ],
     "sqlState" : "0A000"
   },
+  "UNSUPPORTED_SAVE_MODE" : {
+    "message" : [ "unsupported save mode %s" ],
+    "sqlState" : "0A000"
+  },
   "UNSUPPORTED_SIMPLE_STRING_WITH_NODE_ID" : {
     "message" : [ "%s does not implement simpleStringWithNodeId" ]
   },
+  "UNSUPPORTED_STREAMED_OPERATOR_BY_DATA_SOURCE" : {
+    "message" : [ "Data source %s does not support streamed %s" ],
+    "sqlState" : "0A000"
+  },
   "UNSUPPORTED_TRANSACTION_BY_JDBC_SERVER" : {
     "message" : [ "The target JDBC server does not support transaction and can only support ALTER TABLE with a single action." ],
     "sqlState" : "0A000"
   },
+  "WRITING_AMBIGUOUS_DATES" : {
```

Comment:
For these upgrade class names, should we always include the Spark version name? Maybe `WRITING_AMBIGUOUS_NAMES_IN_SPARK3`